src/MacroTools/BookSystem/Powers/PowerBook.cs

```diff
@@ -13,18 +14,17 @@ public sealed class PowerBook : Book<PowerPage>
   {
     private Faction? _trackedFaction;
     private readonly Dictionary<Power, PowerPage> _pagesByPower = new();
-    
+
     /// <summary>
     /// Initializes a new instance of the <see cref="PowerBook"/> class.
     /// </summary>
     /// <param name="trackedPlayer">The player to show <see cref="Power"/>s for.</param>
-    public PowerBook(player trackedPlayer) : base(0.3f, 0.39f, 0.02f, 0.015f)
+    public PowerBook(player trackedPlayer) : base(0.3f, 0.39f, 0.02f, 0.015f, "Powers", new Point(0.4f, 0.36f), 
+      BlzGetFrameByName("UpperButtonBarMenuButton", 0), 
+      trackedPlayer == null || trackedPlayer == GetLocalPlayer())
```

Comment:
This is the only bit I'm concerned about. It's _really_ unclear just from the constructor why this is necessary. I only understand it because I have so much context.

I'm not sure what the solution is but we need to up the readability here. The penalty for getting this wrong is a completely broken game and I don't trust myself to maintain this pattern for future Books.

Reply:
Sorry, to elaborate since I don't think Github's line highlighting made it clear: I'm referring to line 24.

Reply:
Oh I see. I thought you were referring to the constructor having too many parameters and that being what makes this hard to read.

Reply:
Nah, the many parameters is fine. I was just worried about the visibility being determined via a boolean calculation within the constructor like this, and it not being clear to future Book implementers that that's how you achieve a per-player Book. As it stands I think giving that responsibility to the Book Manager is better separation of concerns.